src/main.cc

```diff
@@ -67,22 +67,25 @@ static int status_main(LiteClient& client, const bpo::variables_map& unused) {
   return 0;
 }
 
-static int checkin(LiteClient& client, aklite::cli::CheckMode check_mode) {
+static int checkin(LiteClient& client, aklite::cli::CheckMode check_mode, const bpo::variables_map& params) {
+  bool json_output = false;
+  if (params.count("json") > 0) {
+    json_output = params.at("json").as<bool>();
+  }
+
   std::shared_ptr<LiteClient> client_ptr{&client, [](LiteClient* /*unused*/) {}};
   AkliteClientExt akclient(client_ptr, false, true);
 
-  auto status = aklite::cli::CheckIn(akclient, nullptr, check_mode);
+  auto status = aklite::cli::CheckIn(akclient, nullptr, check_mode, json_output);
   return static_cast<int>(status);
 }
 
-static int cli_list(LiteClient& client, const bpo::variables_map& unused) {
-  (void)unused;
-  return checkin(client, aklite::cli::CheckMode::Current);
+static int cli_list(LiteClient& client, const bpo::variables_map& params) {
+  return checkin(client, aklite::cli::CheckMode::Current, params);
```

Comment:
I think we need to parameterize the `checkin` to indicate whether to do `GetTargetToInstall` or not because the `list` command should just list available targets. There is no need in `GetTargetToInstall` to determine whether a given target is "bad" or "current", only to determine whether it is selected or not. The latter is needed only for the `check` command.

Reply:
As discussed offline, I would rather keep calling `GetTargetToInstall` even for the `list` command, to leave the interface more consistent, and to allow the related functionality to be used even when not updating the TUF metadata.